### PR TITLE
(DEV-3671) Creates prayer feed and prayer component

### PR DESCRIPTION
**Pull request by IsaacHardy**
## DESCRIPTION
This PR consists of the prayer feeds and prayer card within the prayer feeds. 

### What does this PR do, or why is it needed?
This PR is needed to help setup the underlining structure of prayer feeds within the prayer menu.

### How do I test this PR?
Make sure that each prayer feed accurately displays prayers.
---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [DEV-3671](https://newspring.atlassian.net/browse/DEV-3671)
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android if applicable
- [ ] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by cdanwards** [comment]
@IsaacHardy I undid some changes you made to the api. They are included in the PR currently open by @redreceipt and they were breaking the tests. Everything client side is good to go.

File: packages/newspringchurchapp/ios/newspringchurchapp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -298,6 +298,27 @@
 			remoteGlobalIDString = 3D3CD9321DE5FBEE00167DC4;
 			remoteInfo = "cxxreact-tvOS";
 		};
+		425A6CCB22779BF60029BA55 /* PBXContainerItemProxy */ = {
+			isa = PBXContainerItemProxy;
+			containerPortal = 3CFDD4BA5D0E4E2AB58B9F7E /* RNReanimated.xcodeproj */;
+			proxyType = 2;
+			remoteGlobalIDString = 134814201AA4EA6300B7C361;
+			remoteInfo = RNReanimated;
```
Comment (richarddubay) [inline packages/newspringchurchapp/ios/newspringchurchapp.xcodeproj/project.pbxproj]: These items were already added in a prior PR. Not sure why they ended up being changed in here?

File: packages/newspringchurchapp/src/index.js [hunk 2]
```diff
@@ -39,6 +40,7 @@ const AppNavigator = createStackNavigator(
     UserWebBrowser,
     Onboarding,
     Prayer,
+    PrayerList,
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/index.js]: Any reason why this couldn't be added to the PrayerNavigator (on the prayer tab) and not the global navigator. Seems like it would be more appropriate there?

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+const GreyH6 = styled(({ theme }) => ({
+  color: theme.colors.text.tertiary,
+}))(H6);
+
+const StyledCardContent = styled(() => ({
+  alignItems: 'center',
+}))(CardContent);
+
+const StyledBodyText = styled(({ theme }) => ({
+  marginTop: theme.sizing.baseUnit * 1.5,
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]: Per all the other reviews, we should probably avoid putting margin (or other spacing related styles) on actual typography components. Instead, put the margin on wrapping components.

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        return flagRequest(id);
+      }
+      return index;
+    };
+    const options = ['Report Prayer Request', 'Cancel'];
+    return (
+      <StyledCard>
+        <EllipsisView>
+          <ButtonLink onPress={this.handleShowActionSheet}>
+            <GreyH3>...</GreyH3>
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:73]: This works? But maybe an icon would be more appropriate? Not that we have one ... just a thought. FWIW, the H3 looks good.

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          </ButtonLink>
+          <ActionSheet
+            ref={(o) => {
+              this.ActionSheet = o;
+            }}
+            title={'Would you like to remove the prayer request?'}
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]:
```suggestion
            title={'Would you like to report the prayer request?'}
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    const { imageSource, name, text, id, source, flagRequest } = this.props;
+    const cancelIndex = 1;
+    const destructiveIndex = 0;
+    const handleOnPress = (index) => {
+      if (index !== cancelIndex) {
+        return flagRequest(id);
+      }
+      return index;
+    };
+    const options = ['Report Prayer Request', 'Cancel'];
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]:
```suggestion
    const options = ['Report as Inappropriate', 'Cancel'];
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            title={'Would you like to remove the prayer request?'}
+            options={options}
+            cancelButtonIndex={cancelIndex}
+            destructiveButtonIndex={destructiveIndex}
+            onPress={(index) => handleOnPress(index)}
+          />
+        </EllipsisView>
+        <StyledCardContent>
+          <Avatar source={imageSource} size="medium" />
+          <H3> Pray For {name}</H3>
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]:
```suggestion
          <H3>Pray For {name}</H3>
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.stories.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    'I’m alright. Nobody worry ’bout me. Why you got to gimme a fight? Can’t you just let it be?',
+  source: 'Anderson',
+};
+
+storiesOf('PrayerCard', module).add('Example', () => (
+  <PaddedView>
+    <PrayerCard
+      imageSource={cardData.imageSource}
+      name={cardData.name}
+      prayer={cardData.prayer}
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.stories.js]: I think this is now `text` in the `PrayerCard` itself so the text of the prayer no longer displays in Storybook.
```suggestion
      text={cardData.text}
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.stories.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { PaddedView } from '@apollosproject/ui-kit';
+
+import PrayerCard from './PrayerCard';
+
+const cardData = {
+  imageSource: {
+    uri: 'https://fillmurray.com/400/600',
+  },
+  name: 'Bill',
+  prayer:
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.stories.js]: This is `text` in the `PrayerCard`.
```suggestion
  text:
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.tests.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import Providers from 'newspringchurchapp/src/Providers';
+
+import PrayerCard from './PrayerCard';
+
+const cardData = {
+  imageSource: {
+    uri: 'https://fillmurray.com/400/600',
+  },
+  name: 'Bill',
+  prayer:
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.tests.js]: I'm guessing the snapshot for this text is technically broken (even though it won't show it) because `prayer` should be `text`.
```suggestion
  text:
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.tests.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+describe('the PrayerCard component', () => {
+  it('renders a prayer card', () => {
+    const tree = renderer.create(
+      <Providers>
+        <PrayerCard
+          imageSource={cardData.imageSource}
+          name={cardData.name}
+          prayer={cardData.prayer}
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.tests.js]: Same `prayer` to `text` conversion here.
```suggestion
          text={cardData.text}
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+      const currentPrayerList = cache.readQuery({
+        query: getCampusPrayerRequests,
+      });
+      const flagIndex = currentPrayerList.getPublicPrayerRequests.findIndex(
+        (prayer) => prayer.id === flagRequest.id
+      );
+      const incrementFlag = () => {
+        currentPrayerList.getPublicPrayerRequests[flagIndex].flagCount += 1;
+      };
+      const updatedPrayerList = () => {
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.js]: On line 29 you call this function and pass it `(currentPrayerList.getPublicPrayerRequests)`. Do you not need to specify that as a prop to this function? More a question than a critique.

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.tests.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';
+
+import Providers from 'newspringchurchapp/src/Providers';
+import flagPrayerRequest from './flagPrayerRequest';
+import PrayerCardConnected from './PrayerCardConnected';
+
+const prayerId = [
+  {
+    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4789',
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.tests.js]: Is this an Id of an actual prayer in Rock? If so, what would happen if it gets deleted?

File: packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCard.tests.js.snap [hunk 14]
```diff
@@ -0,0 +1,397 @@
...
+                </View>
+              </View>
+            </View>
+          </View>
+        </View>
+      </View>
+    </View>
+  </View>
+</RCTSafeAreaView>
+`;
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCard.tests.js.snap:398]: Yeah, I didn't see the actual prayer text in here.

File: packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCardConnected.tests.js.snap [hunk 15]
```diff
@@ -0,0 +1,481 @@
...
+            style={
+              Object {
+                "color": "#303030",
+                "fontFamily": "Colfax-Black",
+                "fontSize": 24,
+                "lineHeight": 27.6,
+              }
+            }
+          >
+             Pray For
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCardConnected.tests.js.snap]: No name?

File: packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCardConnected.tests.js.snap [hunk 16]
```diff
@@ -0,0 +1,481 @@
...
+                </View>
+              </View>
+            </View>
+          </View>
+        </View>
+      </View>
+    </View>
+  </View>
+</RCTSafeAreaView>
+`;
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/PrayerCard/__snapshots__/PrayerCardConnected.tests.js.snap]: I didn't see any actual prayer text in here either?

**Review by richarddubay — CHANGES_REQUESTED** [review]

File: packages/newspringchurchapp/src/index.js (hunk 2 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/index.js]: I believe this is the result of the nature of react-nav. I think to create the modal-like location, the component needs to exist on this global scale

**Comment by richarddubay** [comment]
When I first load the prayer feed I get the following warnings. Probably shouldn't.
Screenshot 2019-04-30 14 46 53

When I tap on the card to pray for the request I get this error.
Screenshot 2019-04-30 14 47 16

After I flag a request, I get the following warning. 
Screenshot 2019-04-30 14 47 56

Also ... this can wait until after the beta launch, but there needs to be some sort of visual clue that the prayer was flagged. Whether that's removing it from the feed, automatically moving on to the next prayer, or some icon or other visual clue, we need something.

**Comment by IsaacHardy** [comment]
@richarddubay You are 100% correct about the item being removed from feed if flagged. @cdanwards That is probably possible with a optimiseResponse.

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.tests.js (hunk 13 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCardConnected.tests.js]: @richarddubay I think this is fine because this is the dummy data response that we're providing to the mocks.

File: packages/newspringchurchapp/src/index.js (hunk 2 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/index.js]: I added the `AddPrayerForm` to the prayer navigator and it's a modal

Comment (redreceipt) [reply packages/newspringchurchapp/src/index.js]: it definitely needs to go in the prayer navigator, `onClose` takes me back to onboarding.

Kapture 2019-05-01 at 10 17 05

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+
+class PrayerCard extends PureComponent {
+  handleShowActionSheet = () => {
+    this.ActionSheet.show();
+  };
+
+  render() {
+    const { imageSource, name, text, id, flagRequest, source } = this.props;
+
+    const cancelIndex = 1;
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:60]:
```suggestion
    // indices for action sheet buttons
    const cancelIndex = 1;
```

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+    const destructiveIndex = 0;
+    const handleOnPress = (index) => {
+      if (index !== cancelIndex) {
+        return flagRequest(id);
+      }
+      return index;
+    };
+    const options = ['Report as Inappropriate', 'Cancel'];
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:68]: these should really be props so we can add functionality to this card but we can do that in a different PR

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+    return (
+      <StyledCard>
+        <EllipsisView>
+          <ButtonLink onPress={this.handleShowActionSheet}>
+            <GreyH3>...</GreyH3>
+          </ButtonLink>
+          <ActionSheet
+            ref={(o) => {
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]: `o` needs a more descriptive name

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+    const handleOnPress = (index) => {
+      if (index !== cancelIndex) {
+        return flagRequest(id);
+      }
+      return index;
+    };
+    const options = ['Report as Inappropriate', 'Cancel'];
+    return (
+      <StyledCard>
+        <EllipsisView>
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:71]: Don't love the UI on this how the background of the ellipsis turns to gray square

Kapture 2019-05-01 at 10 31 08

File: packages/newspringchurchapp/src/index.js (hunk 2 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/index.js]: That's odd. One of the guys from differential explain this to me and I don't recall what they said. @vinnyjth @conrad-vanl @burkeshartsis "HI!"

Comment (vinnyjth) [reply packages/newspringchurchapp/src/index.js]: Short answer: it depends. 

Long answer: In general if you can access a screen from anywhere in the app (not just a specific part of the app) you should put that screen inside the root navigator. 

However, if you are only using that screen from inside a subsection of the app, you are probably fine putting the route inside that subroute. That's because it's super easy to go "up the tree" for a route, but really hard to go "up and back down". I can explain that more if it doesn't make sense. 

Also, putting a route inside a TabNavigator can have consequences like needing to explicitly hide the tab. Looks like @redreceipt has already solved for that though.

Comment (vinnyjth) [reply packages/newspringchurchapp/src/index.js]: In this case, if prayer list is a modal that can be accessed from anywhere in the app, leave it there. If it's a modal that's only accessed by other screens in the prayer navigator, you can move it back down to the prayer navigator.

**Review by redreceipt — CHANGES_REQUESTED** [review]

File: packages/newspringchurchapp/src/index.js (hunk 2 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/index.js]: @redreceipt I've actually seen that exact behavior on the discover tab in Core Apollos.

Comment (IsaacHardy) [reply packages/newspringchurchapp/src/index.js]: Very odd.

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js (hunk 20 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:71]: We could fix this with a polish PR. I wonder if its caused by us using a sim?

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js (hunk 19 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js]: +1

File: packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js (hunk 20 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/PrayerCard/PrayerCard.js:71]: Agreed. But let's tackle this later. Maybe make a `TODO` to look at it.

**Review by redreceipt — APPROVED** [review]

**Review by richarddubay — APPROVED** [review]
